rules/optimize-android-builds-and-start-up-times/rule.md

```diff
@@ -10,18 +10,114 @@ authors:
+App Bundle [is mandatory for Google Play since 2021](https://developer.android.com/guide/app-bundle).
+
+### Use R8 (instead of ProGuard)
+
+🚫Debug
+✅Release
+
+R8 is a code shrinker and obfuscator which processes **only** native java/kotlin code. It doesn't touch your .NET code.
+As R8 increases build time it's only recommented for Release configuration.
+
+### Use AAPT2 (instead of AAPT)
+
+✅Debug
+✅Release
+
+AAPT2 parses, indexes, and compiles the resources into a binary format that is optimized for the Android platform.
+
+## .NET Layer optimizations
+
+These optimizations are .NET MAUI specific.
+
+### Concurrent GC
+
+✅Debug
+✅Release
+
+Concurrent GC improves app performance by collecting garbage alongside the running program. This approach prevents program from freeze which happen for non-concurrent GCs.
+
+### AOT
+
+🚫Debug
+✅Release
+
+Whereas iOS enforces AOT (ahead of time) compilation, Android supports (and uses by default) JIT (just in time) compilation which happens at runtime, but AOT can be enabled on Android to improve performance. As the code will be pre-compiled, it comes at a cost of significantly larger app size and longer build times.
+There are several types of AOT:
+* Full AOT (comes from Mono)
+* Profiled AOT (comes from Mono, a.k.a Startup Tracing)
+* Native AOT (comes from .NET; not available for Android, but .NET MAUI team is experimenting with iOS)
+
+Profiled AOT provides finer control over the trade-offs between Android APK size and startup time as compared to the Full AOT compilation option.
+Instead of compiling as much of the app as possible to unmanaged code, Profiled AOT compiles only a particular set of managed methods that represent the most expensive parts of application startup in a blank app. This approach consumes less space in the APK compared to the Full AOT compilation option while still providing similar app startup performance improvements.
+
+AOT increases app size and build time (especially Full AOT). Recommended for Release.
+
+### LLVM
+
+🚫Debug
+✅Release
+
+Mono support two compilation engines, a fast, JIT-friendly compilation engine which does not generate very fast code, and a slower compilation engine based on the LLVM optimizing compiler that produces superior code.
+When set LLVM will be used when Ahead-of-Time compiling assemblies into native code.
+
+If Full AOT is not enabled, this property is **ignored**.
+
+Increases build time. Recommended for Release.
+
+### IL Stripping
+
+🚫Debug
+✅Release
+
+$(AndroidStripILAfterAOT) removes IL code after performing AOT as it is no longer required.
+
+### For more information on the topic, see
+
+<https://devblogs.microsoft.com/xamarin/optimize-xamarin-android-builds/>
```

Comment:
Hi @AntPolkanov 

- [ ] 1. Please fix all these links to follow https://www.ssw.com.au/rules/descriptive-links

Reply:
Done